caffe2/proto/caffe2.proto

```diff
@@ -11,27 +11,76 @@ package caffe2;
 //     is mainly for backward compability purposes but we may consider using
 //     those in the future.
 
+// ExternalDataProto stores the pointer to the content of TensorProto
+// Later, we can use it to replace BlobProto
```

Comment:
wait, why would this struct replace BlobProto?

You basically want to serialize a value of arbitrary type. This type might be Tensor or something else. Right now BlobProto serves this purpose. You can merge "custom type" field in TensorProto (as you did in earlier version of this diff) or keep a separate struct. But stuffing it into ExternalData feels out of place